### PR TITLE
Briefly explain how facedir rotations work

### PR DESCRIPTION
**Pull request by aerkiaga**
This PR is just #11918 with two differences:
* No diagrams, in picture or any other form, just a short explanation.
* The Git history got highly messed up, so that's fixed.

## To do

This PR is Ready for Review.

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
Sounds correct.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1069,7 +1069,10 @@ The function of `param2` is determined by `paramtype2` in node definition.
     * Values range 0 - 23
     * facedir / 4 = axis direction:
       0 = y+,   1 = z+,   2 = z-,   3 = x+,   4 = x-,   5 = y-
-    * facedir modulo 4 = rotation around that axis
+    * The node is rotated 90 degrees around the X or Z axis so that its top face
+      points in the desired direction. For the y- direction, it's rotated 180
+      degrees around the Z axis.
+    * facedir modulo 4 = rotation around that axis; always a left-handed rotation.
```
Comment (SmallJoker) [inline doc/lua_api.txt:1075]: What would you think about the following? I believe it's a bit easier to understand, but will let the decision to others.

```
    * facedir / 4 = new direction of the original y+ axis:
      0 = y+,   1 = z+,   2 = z-,   3 = x+,   4 = x-,   5 = y-
    * facedir modulo 4 = left-handed rotation around the specified axis, in 90° steps.
```
This also would solve the special y- direction note that I honestly do not understand.

PS: Confirmed this using the param2 tool in the DevTest game.

Comment (Desour) [reply doc/lua_api.txt:1075]: >This also would solve the special y- direction note that I honestly do not understand.

There are multiple (4) ways how one can rotate one direction to another one.

I.e. you could rotate +y to -y by rotating 180° around the x-axis, the difference would be that one rotation keeps the furnace front face in front (but rotated 180°) and the other one brings the back side to the front.

The details are important if you want to make anything with facedir (see [here](https://github.com/Desour/dslib/blob/71e4d9196d56204534826a10e5bbc97433c3a65d/src/rotnum.lua#L525-L529) if you want to see an example for a mod using this info).
The whole point of this PR is to document facedir, removing the need to read the src, right?

Comment (x2048) [reply doc/lua_api.txt:1075]: I'd still keep the explanation of facedir modulo 4:
```
* facedir modulo 4 = left-handed rotation around the specified axis, in 90° steps.
```

**Comment by Zughy** [comment]
@aerkiaga just one approval missing, please take care of the review so we can proceed :)

**Comment by aerkiaga** [comment]
Sorry for that, it should be done now if I've understood right.

**Review by SmallJoker — APPROVED** [review]
LGTM

**Pull request merged**